qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -231,11 +254,71 @@ def __init__(self,
             get_cmd=":FUNCtion:IMPedance:RANGe?",
             set_cmd=":FUNCtion:IMPedance:RANGe {}",
             unit='Ohm',
-            vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
+            vals=self._imp_range,
             docstring="Selects the impedance measurement range, also turns "
                       "the auto range function OFF."
         )
 
+        self.add_parameter(
+            "dc_bias_enabled",
+            get_cmd=":BIAS:STATe?",
+            set_cmd=":BIAS:STATe {}",
+            vals=Bool(),
+            val_mapping=create_on_off_val_mapping(on_val="1",
+                                                  off_val="0"),
+            docstring="Enables DC bias. DC bias is automatically turned "
+                      "off after recalling the state from memory."
+        )
+
+        self.add_parameter(
+            "dc_bias_voltage_level",
+            get_cmd=":BIAS:VOLTage:LEVel?",
+            set_cmd=":BIAS:VOLTage:LEVel {}",
+            get_parser=float,
+            unit="V",
+            vals=self._dc_bias_v_level_range,
+            docstring="Sets the DC bias voltage. Setting does not "
+                      "implicitly turn the DC bias ON."
+        )
+
+        self.add_parameter(
+            "meas_time_mode",
+            initial_value="MED",
+            vals=Enum("SHOR", "MED", "LONG"),
+            parameter_class=GroupParameter,
+            docstring="Measurement time mode holds the mode of measurement "
+                      "time. It can be SHOR (short), MED (medium - default) "
+                      "or LONG."
+        )
+
+        self.add_parameter(
+            "averaging_rate",
+            initial_value=1,
+            vals=Ints(1, 256),
+            parameter_class=GroupParameter,
+            docstring="Averaging rate for the measurement."
+        )
+
+        self._set_measurement_time = Group(
+            [self.meas_time_mode,
+             self.averaging_rate],
+            set_cmd=":APERture {meas_time_mode},{averaging_rate}",
+            get_cmd=":APERture?"
+        )
+
+        if self.has_firmware_a_02_10_or_above:
+            self.add_parameter(
+                "dc_bias_range_auto",
+                get_cmd=":BIAS:RANGe:AUTO?",
+                set_cmd=":BIAS:RANGe:AUTO {}",
+                vals=Bool(),
+                val_mapping=create_on_off_val_mapping(on_val="1",
+                                                      off_val="0"),
+                docstring="Enables DC Bias range AUTO setting. When DC bias "
+                          "range is FIX (AUTO disabled), '#' is displayed in "
```

Comment:
i'm a bit confused with this docstring. The driver does not have a parameter that specifies the range for the DC bias, so if i do `dc_bias_range_auto(False)`, how will i be able to set the "fixed" range for the DC bias? even, should i be able to set a specific "fixed" range for the DC bias?

Reply:
discussed offline that "FIX" is not something that the instrument "has", it's just capitalized word "fix" (probably should've been fixed, with 'ed'), so all is good here. I would still suggest to reword it a bit like below:
```suggestion
                          "range is fixed (not "AUTO"), '#' is displayed in "
```